### PR TITLE
Split up models.py

### PR DESCRIPTION
**Pull request by amdwit**
draft for splitting up models.py in separate model files.
Let me know what you think of it. 

The contents of User.py are just the first lines of models.py minus the unnecessary imports.

### CODE REVIEW

File: app/backend/src/couchers/models/User.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
+import enum
```
Comment (aapeliv) [inline app/backend/src/couchers/models/User.py]: Please use lowercase filenames in `snake_case`

File: app/backend/src/couchers/models/User.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    naming_convention={
+        "ix": "ix_%(column_0_label)s",
+        "uq": "uq_%(table_name)s_%(column_0_name)s",
+        "ck": "ck_%(table_name)s_%(constraint_name)s",
+        "fk": "fk_%(table_name)s_%(column_0_name)s_%(referred_table_name)s",
+        "pk": "pk_%(table_name)s",
+    }
+)
+
+Base = declarative_base(metadata=meta)
```
Comment (aapeliv) [inline app/backend/src/couchers/models/User.py]: These need to be moved into `__init__.py`, I think

File: app/backend/src/couchers/models/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+from .User import User
```
Comment (aapeliv) [inline app/backend/src/couchers/models/__init__.py]: Please import from `couchers.models.user`, not relative

**Review by aapeliv — DISMISSED** [review]
I think we should split components/groups of models into their own files, not each model separately.

Why is this branch 44 commits behind `develop`? Please rebase onto `develop` and force push.

This is going to be a bit annoying to replicate across both `develop` and `communities`, but probably not too bad if we do the import stuff in `__init__.py` so that there's no change in how dependent code uses models.

File: app/backend/src/couchers/models/__init__.py (hunk 3 above)
Comment (kalvdans) [reply app/backend/src/couchers/models/__init__.py]: I'm not a fan of code in `__init__.py`. Can we leave the file empty please? [EDIT] Oh, I see now that it is to easen merges. I think the benefits of an empty `__init__.py` outweights the extra work resolving merge conflicts.

File: app/backend/src/couchers/models/User.py (hunk 2 above)
Comment (aapeliv) [reply app/backend/src/couchers/models/User.py]: Or to something like `base.py`

File: app/backend/src/couchers/models/__init__.py (hunk 3 above)
Comment (aapeliv) [reply app/backend/src/couchers/models/__init__.py]: OK. I don't see a problem with keeping `__init__.py` clean. In fact, it makes it easier to see where the models really are which is nice.

File: app/backend/src/couchers/models/User.py (hunk 1 above)
Comment (amdwit) [reply app/backend/src/couchers/models/User.py]: fixed

File: app/backend/src/couchers/models/User.py (hunk 2 above)
Comment (amdwit) [reply app/backend/src/couchers/models/User.py]: I moved it to base.py

File: app/backend/src/couchers/models/__init__.py (hunk 3 above)
Comment (amdwit) [reply app/backend/src/couchers/models/__init__.py]: We could empty it (and change imports to it) in a later PR.

**Comment by amdwit** [comment]
Any ideas how to split core.py further?

File: app/backend/src/couchers/models/__init__.py (hunk 3 above)
Comment (kalvdans) [reply app/backend/src/couchers/models/__init__.py]: sounds good to empty it in a later PR :)

**Review by kalvdans — APPROVED** [review]
The suggested file tree structure looks good to me!

**Comment by aapeliv** [comment]
I think the split should be roughly along the same lines as the split of the APIs in `//app/pb`

**Comment by aapeliv** [comment]
Actually, they are quite different, I'm not sure if that makes much sense, and how small/large we want the individual model files to be.

**Comment by amdwit** [comment]
I understand no more changes are needed.

**Comment by aapeliv** [comment]
@amdwit, you did not actually split any of the models into separate files? I do not understand.

**Comment by kalvdans** [comment]
> @amdwit, you did not actually split any of the models into separate files? I do not understand.

That is https://github.com/Couchers-org/couchers/pull/484 and since we are lacking people to finish it I think we should revert this PR, to ease merge of the communities branch. I've opened the revert PR at https://github.com/Couchers-org/couchers/pull/516.

**Pull request merged**